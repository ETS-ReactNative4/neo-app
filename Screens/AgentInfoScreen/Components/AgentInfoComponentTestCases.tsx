import React from "react";

import { ITestCase } from "../../../TypeInterfaces/TestCases/ITestCases";
import AgentPocCard, { IPocCardPropsData } from "./AgentPocCard";
import AgentInfoText from "../../AgentFeedbackScreen/Components/AgentInfoText";
import {
  CONSTANT_passIcon,
  CONSTANT_visaRelatedFaqIcon,
  CONSTANT_paymentIcon
} from "../../../constants/imageAssets";
import AgentInfo from "../AgentInfo";

const pocCardData: IPocCardPropsData[] = [
  {
    title: "Superstar support",
    description: "The travel vouchers you need for your trip",
    iconName: `${CONSTANT_passIcon}`
  },
  {
    title: "Visa assistance",
    description: "The travel vouchers you need for your trip",
    iconName: `${CONSTANT_visaRelatedFaqIcon}`
  },
  {
    title: "Payments",
    description: "The travel vouchers you need for your trip",
    iconName: `${CONSTANT_paymentIcon}`
  }
];

const AgentFeedbackComponentTestCases: ITestCase[] = [
  {
    title: "Agent Info Text Section",
    Component: (
      <AgentInfoText
        agentImage={"https://i.imgur.com/yfA9V3g.png"}
        agentName={"Sunil Sathyaraj"}
        agentDescription={"Your onboarding expert"}
      />
    )
  },
  {
    title: "Agent Poc Card",
    Component: <AgentPocCard pocCardData={pocCardData} />
  },
  {
    title: "Agent Info Screen",
    Component: <AgentInfo />
  }
];

export default AgentFeedbackComponentTestCases;
