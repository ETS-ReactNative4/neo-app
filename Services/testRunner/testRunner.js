import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
