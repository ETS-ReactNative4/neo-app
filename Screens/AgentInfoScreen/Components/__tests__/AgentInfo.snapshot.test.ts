import AgentInfoComponentTestCases from "../AgentInfoComponentTestCases";
import testRunner from "../../../../Services/testRunner/testRunner";

AgentInfoComponentTestCases.map(testRunner);
