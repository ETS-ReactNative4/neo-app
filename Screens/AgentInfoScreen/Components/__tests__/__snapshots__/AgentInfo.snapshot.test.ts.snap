// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agent Info Screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "https://pyt-images.imgix.net/images/place-holder.png",
      }
    }
    style={
      Object {
        "height": 108,
        "left": 0,
        "position": "absolute",
        "resizeMode": "cover",
        "top": 0,
        "width": 205,
      }
    }
  />
  <View
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 100,
          },
        ],
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          undefined,
        ]
      }
    >
      <Image
        onError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://i.imgur.com/yfA9V3g.png",
          }
        }
        style={
          Object {
            "height": 124,
            "marginBottom": 16,
            "width": 124,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "lineHeight": 16,
            "marginBottom": 4,
          }
        }
      >
        Sunil Sathyaraj
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 15,
            "lineHeight": 15,
            "marginBottom": 32,
          }
        }
      >
        Your onboarding expert
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(239, 239, 239, 1)",
            "height": 1,
            "marginBottom": 32,
            "width": 54,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 100,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(44, 47, 66, 1)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 13,
          "lineHeight": 13,
          "marginBottom": 24,
          "textAlign": "center",
        }
      }
    >
      Hello, I’ll be helping you with all of these
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 32,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "rgba(250, 250, 250, 1)",
          "borderRadius": 8,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "padding": 12,
          "transform": Array [
            Object {
              "translateX": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgba(44, 47, 66, 1)",
                "fontSize": 14,
              },
              undefined,
              Object {
                "fontFamily": "voyager",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(44, 47, 66, 1)",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 14,
              "lineHeight": 14,
              "marginBottom": 6,
              "marginTop": 2,
            }
          }
        >
          Superstar support
        </Text>
        <Text
          style={
            Object {
              "color": "rgba(83, 87, 109, 1)",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 13,
              "lineHeight": 13,
            }
          }
        >
          The travel vouchers you need for your trip
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "rgba(250, 250, 250, 1)",
          "borderRadius": 8,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "padding": 12,
          "transform": Array [
            Object {
              "translateX": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgba(44, 47, 66, 1)",
                "fontSize": 14,
              },
              undefined,
              Object {
                "fontFamily": "voyager",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(44, 47, 66, 1)",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 14,
              "lineHeight": 14,
              "marginBottom": 6,
              "marginTop": 2,
            }
          }
        >
          Visa assistance
        </Text>
        <Text
          style={
            Object {
              "color": "rgba(83, 87, 109, 1)",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 13,
              "lineHeight": 13,
            }
          }
        >
          The travel vouchers you need for your trip
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "rgba(250, 250, 250, 1)",
          "borderRadius": 8,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "padding": 12,
          "transform": Array [
            Object {
              "translateX": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgba(44, 47, 66, 1)",
                "fontSize": 14,
              },
              undefined,
              Object {
                "fontFamily": "voyager",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(44, 47, 66, 1)",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 14,
              "lineHeight": 14,
              "marginBottom": 6,
              "marginTop": 2,
            }
          }
        >
          Payments
        </Text>
        <Text
          style={
            Object {
              "color": "rgba(83, 87, 109, 1)",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 13,
              "lineHeight": 13,
            }
          }
        >
          The travel vouchers you need for your trip
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 24,
        "left": 32,
        "opacity": 0,
        "position": "absolute",
        "right": 32,
        "transform": Array [
          Object {
            "translateY": 100,
          },
        ],
        "width": 686,
      }
    }
  >
    <View>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={[Function]}
        style={
          Object {
            "height": 56,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 56,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "rgba(0,198,132,1)",
                "borderRadius": 4,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(247, 247, 247)",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 17,
                  "lineHeight": 17,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`Agent Info Text Section 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onError={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://i.imgur.com/yfA9V3g.png",
      }
    }
    style={
      Object {
        "height": 124,
        "marginBottom": 16,
        "width": 124,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgba(44, 47, 66, 1)",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 16,
        "lineHeight": 16,
        "marginBottom": 4,
      }
    }
  >
    Sunil Sathyaraj
  </Text>
  <Text
    style={
      Object {
        "color": "rgba(83, 87, 109, 1)",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 15,
        "lineHeight": 15,
        "marginBottom": 32,
      }
    }
  >
    Your onboarding expert
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(239, 239, 239, 1)",
        "height": 1,
        "marginBottom": 32,
        "width": 54,
      }
    }
  />
</View>
`;

exports[`Agent Poc Card 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(250, 250, 250, 1)",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 8,
        "padding": 12,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 8,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(44, 47, 66, 1)",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
            "marginBottom": 6,
            "marginTop": 2,
          }
        }
      >
        Superstar support
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 13,
            "lineHeight": 13,
          }
        }
      >
        The travel vouchers you need for your trip
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(250, 250, 250, 1)",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 8,
        "padding": 12,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 8,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(44, 47, 66, 1)",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
            "marginBottom": 6,
            "marginTop": 2,
          }
        }
      >
        Visa assistance
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 13,
            "lineHeight": 13,
          }
        }
      >
        The travel vouchers you need for your trip
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(250, 250, 250, 1)",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 8,
        "padding": 12,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 8,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgba(44, 47, 66, 1)",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "voyager",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
            "marginBottom": 6,
            "marginTop": 2,
          }
        }
      >
        Payments
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(83, 87, 109, 1)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 13,
            "lineHeight": 13,
          }
        }
      >
        The travel vouchers you need for your trip
      </Text>
    </View>
  </View>
</View>
`;
